docs/adr/adr-009-telemetry.md

```diff
@@ -0,0 +1,466 @@
+
+#### Metrics Integration Example
+
+Consider we want to report the current network height as a metric.
+
+First of all, the global pkg meter has to be defined in the code related to the desired metric. In our case it is `header` pkg.
+
+```go
+var meter = global.MeterProvider().Meter("header")
+```
+
+Next, we should understand what instrument to use. On the first glance, for chain height a ___Counter___ instrument should
+fit, as it is a non-decreasing value, and then we should think whether we need a sync or async version of it. For our case,
+both would work and its more the question of precision we want. Sync metering would report every height change, while
+the async would poke `header` pkg API periodically to get the metered data. For our example, we will go with the latter.
+
+```go
+// MonitorHead enables Otel metrics to monitor head.
+func MonitorHead(store Store) {
+ headC, _ := meter.AsyncInt64().Counter(
+  "head",
+  instrument.WithUnit(unit.Dimensionless),
+  instrument.WithDescription("Subjective head of the node"),
+ )
+
+ err := meter.RegisterCallback(
+  []instrument.Asynchronous{
+   headC,
+  },
+  func(ctx context.Context) {
+   head, err := store.Head(ctx)
+   if err != nil {
+    headC.Observe(ctx, 0, attribute.String("err", err.Error()))
+    return
+   }
+
+   headC.Observe(
+    ctx,
+    head.Height,
+    attribute.Int("square_size", len(head.DAH.RowsRoots)),
+   )
+  },
+ )
+ if err != nil {
+  panic(err)
+ }
+}
+```
+
+The example follows a solely API-based approach without the need to integrate the metric deeper into the implementation
+insides, which is nice and keeps metering decoupled from business logic. The `MonitorHead` func simply accepts the `Store`
+interface and reads the information about the latest subjective header via `Head` on the node.
+
+The API-based approach should be followed for any info level metric. Even if there is no API to get the required metric,
+such API should be introduced. However, this approach is not always possible and sometimes deeper integration with code
+logic is necessary to analyze performance or there are security and/or encapsulation considerations.
+
+On the example, we can also see how any additional data can be added to the instruments via attributes or labels. It is
+important to add only absolutely necessary data(more on that in Others section below) to the metrics or data which is
+common over multiple time-series. In this case, we attach `square_size` of the height to know the block size of the
+height. This allows us to query reported heights with some square size using backend UI. Note that there are only
+powers of two(with 256 being a current limit) as unique values possible for the metric, so it won't put pressure on the
+metrics backend.
+  
+For Go code examples on other metric instruments consult [Uptrace Otel docs](https://uptrace.dev/opentelemetry/go-metrics.html#getting-started).
+
+#### Backends Connection
+
+Example for OTLP extracted from our code
+
```

Comment:
This example needs to be updated. Primaraly it misses global attributes we set:
- network name
- node type
- peer ID
Those are crucial for metrics query and filtering and must be a part of adr